### PR TITLE
Minor refactor of LoadFeeTrack (RIPD-956):

### PR DESCRIPTION
**Pull request by ximinez**
* Load scaling functions are fee, and take `Fees`.
* Move LoadFeeTrack to app/misc.
* Update naming convention.

No real functional changes. As described in the ticket, the main objective was to take advantages of `Fees`. Then I did a little clean up while I was there.

### CODE REVIEW

File: src/ripple/app/misc/LoadFeeTrack.h [hunk 1]
```diff
@@ -91,49 +85,67 @@ class LoadFeeTrack
 
     std::uint32_t getLoadFactor () const
     {
-        ScopedLockType sl (mLock);
-        return std::max({ mClusterTxnLoadFee, mLocalTxnLoadFee, mRemoteTxnLoadFee });
+        std::lock_guard <std::mutex> sl (lock_);
+        return std::max({ clusterTxnLoadFee_, localTxnLoadFee_, remoteTxnLoadFee_ });
```
Comment (vinniefalco) [inline src/ripple/app/misc/LoadFeeTrack.h]: Hmm... whats with the curlies?

File: src/ripple/app/misc/LoadFeeTrack.h [hunk 2]
```diff
@@ -91,49 +85,67 @@ class LoadFeeTrack
...
-    beast::Journal m_journal;
-    using LockType = std::mutex;
-    using ScopedLockType = std::lock_guard <LockType>;
-    LockType mutable mLock;
-
-    std::uint32_t mLocalTxnLoadFee;        // Scale factor, lftNormalFee = normal fee
-    std::uint32_t mRemoteTxnLoadFee;       // Scale factor, lftNormalFee = normal fee
-    std::uint32_t mClusterTxnLoadFee;      // Scale factor, lftNormalFee = normal fee
-    int raiseCount;
+    static constexpr std::uint32_t lftNormalFee = 256;        // 256 is the minimum/normal load factor
```
Comment (vinniefalco) [inline src/ripple/app/misc/LoadFeeTrack.h]: I put the `const` or `constexpr` after the type but that's just me

File: src/ripple/app/misc/impl/LoadFeeTrack.cpp [hunk 3]
```diff
@@ -18,54 +18,104 @@
...
-#include <ripple/core/LoadFeeTrack.h>
 #include <ripple/core/Config.h>
+#include <ripple/ledger/ReadView.h>
 #include <ripple/protocol/STAmount.h>
 #include <ripple/protocol/JsonFields.h>
 
 namespace ripple {
 
+bool
+LoadFeeTrack::raiseLocalFee ()
```
Comment (vinniefalco) [inline src/ripple/app/misc/impl/LoadFeeTrack.cpp]: Is this new functionality?

File: src/ripple/app/misc/LoadFeeTrack.h (hunk 1 above)
Comment (nbougalis) [inline src/ripple/app/misc/LoadFeeTrack.h]: That's the new `std::max` that takes an initializer list of `N` items and returns the largest item in the list.

File: src/ripple/app/misc/LoadFeeTrack.h [hunk 4]
```diff
@@ -42,46 +44,38 @@ class LoadFeeTrack
...
-    virtual ~LoadFeeTrack () { }
-
-    // Scale from fee units to millionths of a ripple
-    std::uint64_t scaleFeeBase (std::uint64_t fee, std::uint64_t baseFee,
-        std::uint32_t referenceFeeUnits) const;
-
-    // Scale using load as well as base rate
-    std::uint64_t scaleFeeLoad (std::uint64_t fee, std::uint64_t baseFee,
-        std::uint32_t referenceFeeUnits, bool bUnlimited) const;
+    ~LoadFeeTrack() = default;
```
Comment (scottschurr) [inline src/ripple/app/misc/LoadFeeTrack.h]: Thanks for removing the un-needed `virtual`.  As a sanity check that no one was inheriting from `LoadFeeTrack` I decorated the class declaration (line 43) with `final` and rebuilt successfully.  You could consider leaving the class declaration adorned with `final` as a cheap compile-time check, but it's far from necessary.  Your call.

File: src/ripple/app/misc/LoadFeeTrack.h [hunk 5]
```diff
@@ -42,46 +44,38 @@ class LoadFeeTrack
 {
```
Comment (scottschurr) [inline src/ripple/app/misc/LoadFeeTrack.h]: Is the VFALCO TODO comment on line 42 still useful?

Comment (vinniefalco) [inline src/ripple/app/misc/LoadFeeTrack.h]: We can remove it

File: src/ripple/app/misc/impl/LoadFeeTrack.cpp (hunk 3 above)
Comment (scottschurr) [inline src/ripple/app/misc/impl/LoadFeeTrack.cpp]: Nope.  `raiseLocalFee()` and `lowerLocalFee()` just changed their location the file.

**Comment by scottschurr** [comment]
Looks good to me.  :+1:

**Comment by vinniefalco** [comment]
Seems straightforward
:+1:

File: src/ripple/app/misc/LoadFeeTrack.h (hunk 2 above)
Comment (ximinez) [inline src/ripple/app/misc/LoadFeeTrack.h]: Searching for "static const" vs "static [^ ]* const" (with regexes) finds a nice inconsistent mix (#sarcasm) of both styles, but a huge chunk of the former are in external libraries. I'll change these instances around, but am not going to make any attempt to do a general conversion at this time.

File: src/ripple/app/misc/impl/LoadFeeTrack.cpp (hunk 3 above)
Comment (ximinez) [inline src/ripple/app/misc/impl/LoadFeeTrack.cpp]: No, I (perhaps unnecessarily) moved the converted free functions to after the class functions. `raiseLocalFee` and `lowerLocalFee` are unchanged, other than the class member variable name changes.

File: src/ripple/app/misc/LoadFeeTrack.h (hunk 4 above)
Comment (ximinez) [inline src/ripple/app/misc/LoadFeeTrack.h]: Good point. Added `final`.

File: src/ripple/app/misc/LoadFeeTrack.h (hunk 5 above)
Comment (ximinez) [inline src/ripple/app/misc/LoadFeeTrack.h]: Removed.

**Comment by ximinez** [comment]
I just pushed some trivial changes based on review feedback, but going to mark this as passed anyway.

**Comment by ximinez** [comment]
Rebased to 0.33.0-b1, `expect` converted to `BEAST_EXPECT`, and squashed.

**Comment by nbougalis** [comment]
Merged as d1ce07ef5daf27d6c95079042184e36cfd15eccf

**Pull request merged**